package/src/Slider.tsx

```diff
@@ -216,16 +216,15 @@ const SliderComponent = (
   );
   const [width, setWidth] = useState(0);
 
+  const step = localProps.step
+    ? localProps.step
+    : constants.DEFAULT_STEP_RESOLUTION;
+
   const options = Array.from(
     {
-      length:
-        (localProps.maximumValue! - localProps.minimumValue!) /
-          (localProps.step
-            ? localProps.step
-            : constants.DEFAULT_STEP_RESOLUTION) +
-        1,
+      length: (localProps.maximumValue! - localProps.minimumValue!) / step + 1,
```

Comment:
This is the only thing that bothers me in this PR: What if minimumValue or maximumValue are not given? We have the default ones, maybe we should use those defaults here?

Reply:
I have to ask what do you mean under the default `minimumValue` and `maximumValue`? The value which are defined under the `SliderWithRef.defaultProps `? 

I removed my defined max and min values and if I defined the step as 1 the custom marker works as expected. 

Code from the modified Example code:

```js
const SliderExampleWithCustomMarker = (props: SliderProps) => {
  const [value, setValue] = useState(props.value ?? CONSTANTS.MIN_VALUE);

  return (
    <View>
      <Text style={styles.text}>{value && +value.toFixed(3)}</Text>
      <Slider
        step={1}
        style={[styles.slider, props.style]}
        thumbImage={require('./resources/slider-left.png')}
        renderStepNumber
        tapToSeek
        {...props}
        value={value}
        onValueChange={setValue}
        StepMarker={MyStepMarker}
        minimumTrackTintColor={'#00629A'}
        maximumTrackTintColor={'#979EA4'}
      />
    </View>
  );
};
```

<img src="https://github.com/callstack/react-native-slider/assets/19413170/1ff2ac1c-b998-43d0-bd02-2f9dafdc63bd" width="300" />
<img src="https://github.com/callstack/react-native-slider/assets/19413170/594a8bae-f4e8-4314-a4a9-49a229d9405c" width="300" />

However unfortunately if the step is the default one (100 as DEFAULT_STEP_RESOLUTION) my logic is broken. Also if the step is for example 0.2 because the division. Do you have any suggestion how we can resolve that?

Reply:
@betko ☝️ (forgot to mention/ping)